### PR TITLE
Clean up warnings

### PR DESCRIPTION
**Pull request by twsouthwick**
This does add back rules.ruleset since .editoconfig does not work for files outside of the project

### CODE REVIEW

**Review by mjrousos — CHANGES_REQUESTED** [review]
Thanks for fixing these up, @twsouthwick! I left a few comments - mostly about the placement of the shared source file with the CLSCompliant attribute. Otherwise, this looks good.

**Comment by mjrousos** [comment]
Looks like something broke in the TryConvertStepTests.InitializeTests tests.

**Comment by mjrousos** [comment]
I fixed the test issue. It turns out that one of the fixes I made (making the catch in `TryConvertProjectConverterStep` catch a more specific MSBuild exception) introduced a test failure because it meant that the TryConvert step tests needed to initialize MSBuild. But there's really no reason for those tests to need an MSBuild dependency, so I just reverted the change to `TestConvertProjectConverterStep`.

File: .editorconfig [hunk 1]
```diff
@@ -255,6 +255,9 @@ dotnet_diagnostic.SA1310.severity = none
 # SA1502: Element should not be on a single line
 dotnet_diagnostic.SA1502.severity = none
 
+# CA1062: Validate arguments of public methods
+dotnet_diagnostic.CA1062.severity = none
```
[the remark this answers is not in the document]
Comment (twsouthwick) [reply .editorconfig:259]: I was thinking it's public, but we're building everything in a nullable context. However, with extension possibilities, probably do want that.

File: Directory.Build.targets [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+
   <ItemGroup>
     <PackageReference Include="StyleCop.Analyzers">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)/common/**/*.cs" />
```
[the remark this answers is not in the document]
Comment (twsouthwick) [reply Directory.Build.targets:11]: But we need this in the test directory as well, so it seems to be better to have top level, but I'm happy to change it. I like the shared name though

File: .editorconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (twsouthwick) [reply .editorconfig]: Done

File: rules.ruleset [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<RuleSet Name="Rules Upgrade Assistant" ToolsVersion="15">
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+     <!-- Required for https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3092 -->
+     <Rule Id="SA0001" Action="None" />
+     <Rule Id="SA1633" Action="None" />
```
[the remark this answers is not in the document]
Comment (twsouthwick) [reply rules.ruleset:5]: I'd rather not have to keep that in sync, though

File: Directory.Build.targets (hunk 2 above)
[the remark this answers is not in the document]
Comment (twsouthwick) [reply Directory.Build.targets]: Switched the name. Let me know if you still think we should move the location

**Comment by twsouthwick** [comment]
We may want to go through and figure out the exceptions that we would need to catch, so we don't have to disable the warning, but going forward new exceptions will be flagged

File: Directory.Build.targets (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjrousos) [reply Directory.Build.targets]: Ah, I didn't think about the tests. Ok, I'm good with this.

**Review by mjrousos — APPROVED** [review]

**Pull request merged**